File: Programmers/Lv.1/택배 상자 꺼내기/389478.js
function solution(n, w, num) {
    let answer = 0;
    
    while (num <= n) {
        const x = parseInt( (num - 1) / w );
        const position = (num - 1) % w;
        const y = (x % 2 === 0) ? position : w - position - 1;
        
        if (x % 2 === 0) {
            num = num + (w - y) + (w - y - 1);
        } else {
            num = num + y + y + 1;
        }
        
        answer++;
    }
    
    return answer;
}

File: Programmers/Lv.1/택배 상자 꺼내기/389478.java
import java.util.*;

class Solution {
    public int solution(int n, int w, int num) {
        int answer = 0;
        
        while (num <= n) {
            int x = (num - 1) / w;
            int position = (num - 1) % w;
            int y = (x % 2 == 0) ? position : w - position - 1;
        
            if (x % 2 == 0) {
                num = num + (w - y) + (w - y - 1);
            } else {
                num = num + y + y + 1;
            }
            
            answer++;
        }
        
        return answer;
    }
}
